specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json

```diff
@@ -8045,8 +8048,11 @@
         },
         "headers": {
           "type": "object",
-          "x-ms-format": "dfe-key-value-pairs",
-          "description": "Represents the headers that will be sent to the request. For example, to set the language and type on a request: \"headers\" : { \"Accept-Language\": \"en-us\", \"Content-Type\": \"application/json\" }. Type: dictionary (or Expression with resultType dictionary)."
+          "description": "Represents the headers that will be sent to the request. For example, to set the language and type on a request: \"headers\" : { \"Accept-Language\": \"en-us\", \"Content-Type\": \"application/json\" }. Type: string (or Expression with resultType string).",
+          "additionalProperties": {
```

Comment:
>additionalProperties [](http://example.com/codeflow?start=11&length=20)

Use of additionalProperties is not allowed for properties owned by the service. The only time its ok to use it is when the properties are pass thru (user defined) and not subject to any validations. Please schematize this object, make it an array or provide an explanation for why you need to use this pattern.

Reply:
The additionalProperties we've included serve as HTTP headers specified by the customer. This schema does not follow a specific pattern.

Moreover, the ADF SDK has some unique x-ms-format, identified by a 'dfe-' prefix, such as 'dfe-key-value-pairs'. This update in the schema also addresses an issue in the .NET Track2 SDK, where customers were unable to create web activities with headers (refer to [Azure SDK for .NET Issue #39187](https://github.com/Azure/azure-sdk-for-net/issues/39187) for more details). We've conducted local tests, and the solution appears to be functioning effectively.

Reply:
Are the HTTP headers just passed straight through without any examination or validation applied?

Reply:
Yes.  The HTTP headers just passed through. No validation.